code/addons/viewport/src/Tool.tsx

```diff
@@ -106,11 +106,25 @@ const getStyles = (
   styles: Styles,
   isRotated: boolean
 ): ViewportStyles | undefined => {
-  if (!styles || !prevStyles) {
+  if (styles === null) {
     return undefined;
   }
-  const result = typeof styles === 'function' ? styles(prevStyles) : styles;
-  return isRotated ? flip(result) : result;
+
+  let result: ViewportStyles | undefined;
+
+  if (typeof styles === 'function') {
+    if (prevStyles) {
+      result = styles(prevStyles);
+    }
+  } else {
+    result = styles;
+  }
```

Comment:
But what now if styles is a function and prevStyles is not yet defined. Then `result` will also be always undefined right? I think that maybe the right fix would be to rollback what we had before, and change the type to be:

```ts
export type Styles = ViewportStyles | ((previous: ViewportStyles | null) => ViewportStyles) | null;
```
So that the user has to explicitly handle previous being still `null`.

Reply:
> to rollback what we had before

You mean reverting back to before fe02579, right?

And yes you're right the nested if `if (prevStyles)` was only there because of the type `Styles`.

And what about the type for `prevStyles`? The actual value for `prevStyles` argument comes from `const ref = useRef<ViewportStyles>();` which allows it to be `undefined`, so you either have to give it the initial value or accept `undefined` for `prevStyles` arg of `getStyles`.

Reply:
> You mean reverting back to before https://github.com/storybookjs/storybook/commit/fe02579b4ca5f162044ab5ca28293228458fc28f, right?

Yes!

Oh wait, widening `previous` in the type of `Styles` to `undefined` instead of `null` sounds also good to me indeed!